python/pyspark/sql/tests/connect/test_connect_basic.py

```diff
@@ -829,6 +829,13 @@ def test_with_columns(self):
             .toPandas(),
         )
 
+    def test_hint(self):
+        # SPARK-41349: Test hint
```

Comment:
Please add additional tests for:

  * unsupported param types
  * unsupported hint name
  * invalid combination of hint and param

Please check if there is additional coverage needed.

Reply:
Good catch, I added 4 more test cases for it.